libs-scala/scala-utils/src/main/scala/com/daml/scalautil/nonempty/NonEmpty.scala

```diff
@@ -43,17 +42,17 @@ sealed abstract class NonEmptyColl {
     */
   def equiv[F[_], A]: NonEmpty[F[A]] === NonEmptyF[F, A]
 
-  /** Check whether `self` is non-empty; if so, return it as the non-empty subtype. */
-  @deprecated("misleading apply, use `case NonEmpty(xs)` instead", since = "1.18.0")
-  final def apply[Self](self: Self with imm.Iterable[_]): Option[NonEmpty[Self]] = unapply(self)
-
   /** {{{
-    *  NonEmpty.pour(1, 2, 3) into List : NonEmpty[List[Int]] // with (1, 2, 3) as elements
+    *  NonEmpty(List, 1, 2, 3) : NonEmpty[List[Int]] // with (1, 2, 3) as elements
     * }}}
-    *
-    * The weird argument order is to support Scala 2.12.
     */
-  final def pour[A](x: A, xs: A*): Pouring[A] = new Pouring(x, xs: _*)
+  final def apply[Fct, A, C <: imm.Iterable[A]](into: Fct, hd: A, tl: A*)(implicit
+      fct: Fct => Factory[A, C]
+  ): NonEmpty[C] = {
```

Comment:
We _still_ have to delay selection of the conversion to `Factory` just as `pour` did. If we don't:

```scala
scala> def two[A, C](fct: collection.Factory[A, C], a: A) = fct
def two[A, C](fct: scala.collection.Factory[A,C], a: A): scala.collection.Factory[A,C]

scala> two(List, 1)
           ^
       error: type mismatch;
        found   : scala.collection.Factory[Nothing,List[Nothing]]
        required: scala.collection.Factory[1,List[Nothing]]
```